src/libsync/discovery.cpp

```diff
@@ -230,25 +182,41 @@ void ProcessDirectoryJob::process()
 #endif
         if (handleExcluded(path._target, e.localEntry.name,
                 e.localEntry.isDirectory || e.serverEntry.isDirectory, isHidden,
-                e.localEntry.isSymLink || isServerEntryWindowsShortcut))
+                e.localEntry.isSymLink || isServerEntryWindowsShortcut, e.serverEntry.isValid() || e.dbEntry.isValid()))
             continue;
 
         if (_queryServer == InBlackList || _discoveryData->isInSelectiveSyncBlackList(path._original)) {
             processBlacklisted(path, e.localEntry, e.dbEntry);
             continue;
         }
-        if (!checkForInvalidFileName(path, entries, e)) {
-            continue;
-        }
         processFile(std::move(path), e.localEntry, e.serverEntry, e.dbEntry);
     }
     QTimer::singleShot(0, _discoveryData, &DiscoveryPhase::scheduleMoreJobs);
 }
 
-bool ProcessDirectoryJob::handleExcluded(const QString &path, const QString &localName, bool isDirectory, bool isHidden, bool isSymlink)
+bool ProcessDirectoryJob::handleExcluded(const QString &path, const QString &localName, bool isDirectory, bool isHidden, bool isSymlink, bool isNotJustCreated)
```

Comment:
This is a _very_ small nitpick but `isNotJustCreated` isn't very descriptive/clear in what it actually is referencing and it is a negative -- something like `isInLocalOrRemoteDb` more clearly says what this argument is used for